src/java/org/apache/cassandra/db/commitlog/CommitLogDescriptor.java

```diff
@@ -63,13 +63,14 @@ public class CommitLogDescriptor
     public static final int VERSION_30 = 6;
     public static final int VERSION_40 = 7;
     public static final int VERSION_50 = 8;
+    public static final int VERSION_51 = 9;
 
     /**
      * Increment this number if there is a changes in the commit log disc layout or MessagingVersion changes.
      * Note: make sure to handle {@link #getMessagingVersion()}
      */
     @VisibleForTesting
-    public static final int current_version = DatabaseDescriptor.getStorageCompatibilityMode().isBefore(5) ? VERSION_40 : VERSION_50;
+    public static final int current_version = DatabaseDescriptor.getStorageCompatibilityMode().isBefore(5) ? VERSION_40 : VERSION_51;
```

Comment:
I think that we do need a ` CASSANDRA_5(5)` compatibility mode otherwise mutations in the commitlog  and hints will contain epochs so downgrading to 5.0 will not work unless the 5.1 instance uses SCM `CASSANDRA_4`. It may not be possible to do this if it previously wrote 5.0 format sstables, so a downgrade to 5.0 would only work if the cluster had _always_ been in SCM 4.

Of course, we could punt downgradability & a new SCM mode to a later ticket. This versioning bump is needed now though to prevent 5.1 nodes including epochs in messages to 5.0 nodes.

Reply:
absolutely!